### ISSUE TITLE
sam_hdr_parse does not assign 'text' and 'l_text'

### ISSUE THREAD

**Issue by nh13** [samtools/htslib#104]
This will seg-fault:

```
        h = sam_hdr_parse(str.l, str.s);
        sam_hdr_write(out, h);
```

This will not seg-fault:

```
        h = sam_hdr_parse(str.l, str.s);
        h->l_text = str.l; h->text = str.s;
        sam_hdr_write(out, h);
```

**Comment by kirkmcclure** [comment]
The structure is initialized by calloc() in bam_hdr_init(), so the problem is solely in sam_hdr_write() where a guard condition belongs - maybe:
````
int sam_hdr_write(htsFile *fp, const bam_hdr_t *h)
{
    if ( h == NULL || h->text == NULL ) { // htslib Issue #104
        fprintf(stderr, "[E:%s] invalid header\n", __func__);
        return -2;
    }
...
````

**Comment by SamStudio8** [comment]
#293

**Comment by valeriuo** [comment]
Fixed by #812
